File: src/handlers/error_handler.py
import sys

sys.path.append(sys.path[0][: -len("handlers")] + "utils")
from messages import messages
from show import show


def error_handler(callback, close=None):
    try:
        callback()
    except Exception as err:
        show("error.txt", [str(err)])
        if str(err) == messages["throw"]:
            if close:
                close()

File: src/handlers/error-handler.js
import { messages } from "../utils/messages.js";
import { show } from "../utils/show.js";

export const errorhandler = async (callback, close) => {
  try {
    await callback();
  } catch (err) {
    await show("error.txt", [err.message]);
    if (err.message === messages.throw) {
      close && close();
    }
  }
};
